src/estimators/any_sketch.py

```diff
@@ -47,6 +47,40 @@ def __call__(self, x, y):
     return x | y
 
 
+class UniqueKeyFunction(ValueFunction):
+  """ValueFunction to track the state of unique key of a register."""
+
+  FLAG_EMPTY_REGISTER = -1
+  FLAG_COLLIDED_REGISTER = -2
+
+  def __call__(self, x, y):
+    """ValueFunction to track the state of unique key of a register.
+
+    Args:
+      x: A state of unique key. It can be either a real key (hashed ID)
+        indicating the unique key in the register, or FLAG_EMPTY_REGISTER
+        indicating that the register is empty, or FLAG_COLLIDED_REGISTER
+        indicating that the register already has collision.
+      y: Another state of unique key.
+
+    Returns:
+      A state of unique key that merges x and y.
+    """
+    if x == UniqueKeyFunction.FLAG_EMPTY_REGISTER:
+      return y
+    if y == UniqueKeyFunction.FLAG_EMPTY_REGISTER:
+      return x
+    if (x == UniqueKeyFunction.FLAG_COLLIDED_REGISTER) or (
```

Comment:
You don't need the parens around each of the terms of the "or".

Reply:
Thanks, removed.